src/zjs_fs.c

```diff
@@ -409,19 +413,27 @@ static jerry_value_t zjs_read(const jerry_value_t function_obj,
         return zjs_error("offset + length overflows buffer");
     }
 
-    // if a position was specified, seek to it before writing
-    if (fs_seek(&handle->fp, (uint32_t)position, SEEK_SET) != 0) {
+    if (handle->mode == MODE_A_PLUS) {
+        // mode is a+, seek to read position
+        if (fs_seek(&handle->fp, handle->rpos, SEEK_SET) != 0) {
+            return zjs_error("error seeking to position");
+        }
+    }
+    // if a position was specified, seek to it before reading
+    if (fs_seek(&handle->fp, (uint32_t)position, SEEK_CUR) != 0) {
```

Comment:
This ignores the seek you maybe just did above. Also I think this should be SEEK_SET instead of SEEK_CUR, the position passed in I think is an absolute position from the beginning of the file, not from the current point.

Also, according to node docs position can be "null" and that means just use the current position. So you ought to check for that I guess; from this comment it seems you intended to?